aptos-move/framework/aptos-names/sources/verify.move

```diff
@@ -0,0 +1,34 @@
+module aptos_names::verify {
+    use std::string;
+    use std::vector;
+    use aptos_framework::chain_id;
+    use aptos_std::ed25519;
+    use aptos_names::config;
+
+    friend aptos_names::domains;
+
+    struct RegisterDomainProofChallenge has drop {
+        sequence_number: u64,
+        register_address: address,
+        domain_name: string::String,
+        chain_id: u8,
+    }
```

Comment:
@alinush , if we're going this direction, do we want to make this a one off as it currently stands, or do you think the following is sufficient API for domain separation:

```
MOVE_PREFIX | struct address | struct module | struct name | chain_id | struct data
```

if so, the API might be better defined as making a new verifier API that would wrap the proof challenge in a domain separated struct. E.g.,

```
struct DomainSeparated<T> {
  prefix: vec<u8>,
  chain_id: u8,
  data: T,
}
```

where 
```
struct DomainSeparated {
  prefix: b"SAFE_MOVE",
  chain_id: 1,
  data: RegisterDomainProofChallenge {
    sequence_number: 5,
    registrant_address: @0x05,
    domain_name: String(b"david"),
  }
}
```

the chain_id can be retrieved without the application layer even knowing!!!

I reckon there is no collision with anything in the ecosystem.

Reply:
After talking to Alin, my understanding was `MOVE_PREFIX | struct address | struct module | struct name` should be SHA3_256'd first, and then the rest together

Reply:
It is _not_ sufficient: we need to consider the interaction with the wallet software that's asking users to sign things. Specifically, the software needs to display the struct being signed field-by-field, including field names, types and contents (and proceed recursively); not just display the serialized blob of data being signed.

In other words, we'll need something similar to [EIP-712](https://eips.ethereum.org/EIPS/eip-712) which signs not only the contract's address, struct name and its contents, but also the struct's type information (recursing all the way down to primitive types). 

@CapCap showed me on Friday how we might do this in Move, so you just have to sit tight and stick to our current domain separation via `ed25519::verify_signature_strict_t`  until the standard is ready.

Reply:
That being said, including the chain ID in `RegisterDomainProofChallenge` is a good idea!